engine/src/test/scala/cromwell/services/KeyValueServiceActorSpec.scala

```diff
@@ -0,0 +1,186 @@
+package cromwell.services
+
+import akka.pattern.ask
+import akka.util.Timeout
+import com.typesafe.config.ConfigFactory
+import cromwell.CromwellTestkitSpec.TestWorkflowManagerSystem
+import cromwell.engine.WorkflowSourceFiles
+import cromwell.engine.backend.local.LocalBackend
+import cromwell.engine.db.slick.SlickDataAccess
+import cromwell.services.KeyValueServiceActor._
+import org.scalatest.{FlatSpec, Matchers}
+
+import scala.concurrent.Await
+import scala.concurrent.duration._
+
+class KeyValueServiceActorSpec extends FlatSpec with Matchers with CromwellServicesTest {
```

Comment:
Mix in `BeforeAndAfterAll`, then clean up after yourself. Each test like this that leaks an actor system (below) is leaving behind a pool of threads among other resources.

```scala
  override protected def afterAll() = {
    testWorkflowManagerSystem.shutdownTestActorSystem()
    super.afterAll()
  }
```